backend_py/libs/services/src/webviz_services/summary_vector_statistics.py

```diff
@@ -41,77 +41,54 @@ def compute_vector_statistics_table(
+            statistics_expressions.append(valid_col_expr.mean().alias("MEAN"))
         elif stat_func == StatisticFunction.P10:
-            agg_dict["P10"] = pd.NamedAgg(column=vector_name, aggfunc=p10_func)
+            # Inverted due to oil industry convention (P10 = 90th percentile)
+            statistics_expressions.append(valid_col_expr.quantile(0.9).alias("P10"))
         elif stat_func == StatisticFunction.P90:
-            agg_dict["P90"] = pd.NamedAgg(column=vector_name, aggfunc=p90_func)
+            # Inverted due to oil industry convention (P90 = 10th percentile)
+            statistics_expressions.append(valid_col_expr.quantile(0.1).alias("P90"))
         elif stat_func == StatisticFunction.P50:
-            agg_dict["P50"] = pd.NamedAgg(column=vector_name, aggfunc=p50_func)
-
-    if not agg_dict:
-        raise InvalidParameterError("At least one statistic must be requested", Service.GENERAL)
-
-    if summary_vector_table.num_rows == 0:
-        return None
-
-    df = summary_vector_table.select(["DATE", vector_name]).to_pandas(timestamp_as_object=True)
+            statistics_expressions.append(valid_col_expr.quantile(0.5).alias("P50"))
 
-    grouped: pd.core.groupby.DataFrameGroupBy = df.groupby("DATE", as_index=False, sort=True)
-    statistics_df: pd.DataFrame = grouped.agg(**agg_dict)
+    # Create Polars DataFrame from Arrow table and compute statistics
+    vector_df = pl.DataFrame(summary_vector_table.select(["DATE", vector_name]))
+    statistics_df = vector_df.sort("DATE").group_by("DATE", maintain_order=True).agg(statistics_expressions)
 
-    default_schema = pa.Schema.from_pandas(statistics_df, preserve_index=False)
-    schema_to_use = set_date_column_type_to_timestamp_ms(default_schema)
-    schema_to_use = create_float_downcasting_schema(schema_to_use)
+    # Convert to PyArrow
+    statistics_table = statistics_df.to_arrow()
 
-    statistics_table = pa.Table.from_pandas(statistics_df, schema=schema_to_use, preserve_index=False)
+    # Downcast float columns to save memory
+    schema_to_use = create_float_downcasting_schema(statistics_table.schema)
+    statistics_table = statistics_table.cast(schema_to_use)
 
     return statistics_table
 
 
 def compute_vector_statistics(
     summary_vector_table: pa.Table,
     vector_name: str,
-    statistic_functions: Optional[Sequence[StatisticFunction]],
-) -> Optional[VectorStatistics]:
+    statistic_functions: Sequence[StatisticFunction] | None,
+) -> VectorStatistics | None:
     statistics_table = compute_vector_statistics_table(summary_vector_table, vector_name, statistic_functions)
     if not statistics_table:
         return None
 
-    unique_realizations: List[int] = []
+    unique_realizations: list[int] = []
     if "REAL" in summary_vector_table.column_names:
         # ! We assume the list never has None-values
         unique_realizations = cast(list[int], summary_vector_table.column("REAL").unique().to_pylist())
```

Comment:
Again, probably irrelevant here, but we use to_pylist all over the place.
We should probably use to_numpy().tolist()
https://github.com/apache/arrow/issues/34354

Reply:
Okay, I wasn't aware of that detail. I made a search and replace other places where `to_pylist()` was used as well.